python/gherkin3/token_matcher.py

```diff
@@ -17,6 +20,41 @@ def reset(self):
         self._indent_to_remove = 0
         self._active_doc_string_separator = None
 
+    def _change_dialect(self, dialect_name, location=None):
```

Comment:
There are different strategies for ordering functions and methods. Public methods before private methods is one - it was basically the one used in this class previously - even though `_match_DocStringSeparator` was an exception to that rule. "In call graph order" is another strategy (like `_match_DocStringSeparator` follows `match_DocStringSeparator` from which is called).
Here I do not see the strategy used. It is not "methods should be declared before they are used", because `_change_dialect` is placed after `reset` which calls it.

The main objection I have is the move of `_unescaped_docstring` it is only used in `match_Other`, so I think it should be kept closer to that method.